engine/step/step.go

```diff
@@ -384,8 +390,18 @@ func Run(st *Step, baseConfig map[string]json.RawMessage, stepValues *values.Val
 		// Wait the prehook execution is done
 		preHookWg.Wait()
 
+		// after pre-hook execution, let's verify if we are not shutting down the instance
+		// in that case, we can just put the step in ToRetry instead of starting the main execution of the step
+		select {
+		case <-shutdownCtx.Done():
+			st.State = StateToRetry
+			go noopStep(st, stepChan)
```

Comment:
Shoudn't use a synchronous call of `noopStep` instead ? You're firing a goroutine while the instance is shutting down, that goroutine won't be accounted for and main process won't wait for it to finish I think.